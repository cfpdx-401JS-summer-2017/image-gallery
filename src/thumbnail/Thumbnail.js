import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Thumbnail extends Component {

    static propTypes = {
        bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }

    render() {
        return (
            <div className="thumbnails">
                { this.props.bunnies.map( ( bunny, index ) => {
                    return (
                        <div key={index}>
                            <img src={bunny.url} alt={bunny.title}/>
                            <h6>{bunny.title}</h6>
                        </div>
                    )
                })}
            </div>
        )
    }

}