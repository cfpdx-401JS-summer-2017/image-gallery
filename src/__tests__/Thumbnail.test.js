import React from 'react';

import { shallow } from 'enzyme';

import toJSON from 'enzyme-to-json';
import Bunnies from '../images/bunnies';
import Thumbnail from '../thumbnail/Thumbnail';

describe( 'Thumbnail', () => {

    it( 'lists bunnies as thumbnail photos', () => {
        const wrapper = shallow(<Thumbnail bunnies={Bunnies}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});