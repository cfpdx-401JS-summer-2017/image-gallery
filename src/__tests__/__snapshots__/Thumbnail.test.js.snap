// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail lists bunnies as thumbnail photos 1`] = `
<div
  className="thumbnails"
>
  <div>
    <img
      alt="Cute Bunny"
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    />
    <h6>
      Cute Bunny
    </h6>
  </div>
  <div>
    <img
      alt="Cuter Bunny"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
    />
    <h6>
      Cuter Bunny
    </h6>
  </div>
  <div>
    <img
      alt="Cutest Bunny"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
    />
    <h6>
      Cutest Bunny
    </h6>
  </div>
</div>
`;
